### PR TITLE
Rework virt tests

### PR DESCRIPTION
**Pull request by coolo**
Requires https://github.com/os-autoinst/os-autoinst/pull/789 and test setup changes

### CODE REVIEW

**Comment by coolo** [comment]
CC @alice-suse - I wasn't able yet to test this completely, but this is the idea

**Comment by alice-suse** [comment]
Thanks a lot for it. I will have a look at it, and if you need anything to do by me, just ask.

File: tests/installation/installation_overview.pm [hunk 1]
```diff
@@ -45,7 +45,10 @@ sub run() {
 
     $self->check_and_record_dependency_problems;
 
-    if (check_var('ARCH', 's390x') && !get_var('UPGRADE')) {    # s390x always needs SSH
+    my $need_ssh = check_var('ARCH', 's390x');    # s390x always needs SSH
+    $need_ssh = 1 if check_var('BACKEND', 'ipmi');    # we better be able to login
+
+    if (!get_var('UPGRADE') && $need_ssh) {
```
Comment (asmorodskyi) [inline tests/installation/installation_overview.pm]: if ((check_var('ARCH', 's390x') && !get_var('UPGRADE')) ||  check_var('BACKEND', 'ipmi') )
this form looks more clear for me

**Comment by alice-suse** [comment]
@coolo , I read through the code. Would you please give a short description about the main changes the test code in virtualization needs? 

From my understanding, I get following:
Need change:
1) the start point after each 'reboot' is different, so test flow should adjust to it and collect new needles
2) after every login, use root-ssh console instead of root console
3) set up SERIALDEV to sshserial for serial console setting. But here I have a question. As you know we have special code for kvm and xen different serial console setting, both grub setting and serialdev parameter. After introducing sshserial, what the serial setting should be in grub, especially on xen?
4) before reboot, need to switch back to ipmi console. Then how? By selecting root console?  Are both ssh and ipmi connections kept at the same time in the new backend?
5) SUT-IP is dynamically assigned by DHCP, so need code to set it up, rather than given it statically in testsuite setting


Consistent:
1) test apis

**Comment by coolo** [comment]
3. SERIALDEV is the device we get the wait_serial stuff from - this should always be sshserial - no matter if this is xen or kvm. No need to play with the grub config as you won't see the output of the console anyway, serial0 will contain only what is written to sshserial. And this removes the necessarity of reboots anyway

4. before reboots you need to take care to disable the root-ssh console and select_console 'sol'. In case this doesn't work, let me know.

5. SUT_IP can also take a DHCP hostname. But we set all SUTs to susetest in DHCP so that needles are consistent, so we have to take the IP.

**Comment by alice-suse** [comment]
Get it. Thanks. I will work on it.

File: tests/installation/installation_overview.pm (hunk 1 above)
Comment (coolo) [inline tests/installation/installation_overview.pm]: not to me

**Comment by coolo** [comment]
Merging this to get the sync up with the backend changes - virt tests are broken atm, so improving this is the main objective anyway

**Pull request merged**